src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1491,9 +1496,28 @@ private void AddMemberLookupSymbolsInfoInSubmissions(LookupSymbolsInfo result, T
         {
             // TODO: we need tests
             // TODO: optimize lookup (there might be many interactions in the chain)
+
             for (CSharpCompilation submission = Compilation; submission != null; submission = submission.PreviousSubmission)
             {
-                submission.GetSubmissionImports().AddLookupSymbolsInfoInAliases(this, result, options);
+                var usingAliases = submission.SubmissionUsingAliases?.Values;
+                if (usingAliases != null)
+                {
+                    foreach (var usingAlias in usingAliases)
+                    {
+                        var aliasSymbol = usingAlias.Alias;
+                        var targetSymbol = aliasSymbol.GetAliasTarget(basesBeingResolved: null);
+                        if (originalBinder.CanAddLookupSymbolInfo(targetSymbol, options, null))
+                        {
+                            // Note: we do not mark nodes when looking up arities or names.  This is because these two
+                            // types of lookup are only around to make the public
+                            // SemanticModel.LookupNames/LookupSymbols work and do not count as usages of the directives
+                            // when the actual code is bound.
+
+                            result.AddSymbol(aliasSymbol, aliasSymbol.Name, 0);
+                        }
+                    }
```

Comment:
This loop appears three times. Consider extracting body to AddToResult(originalBinder, result, options, alias) method.